decoder/vaapidecoder_vp9.cpp

```diff
@@ -90,23 +91,50 @@ void VaapiDecoderVP9::flush(void)
 
 Decode_Status VaapiDecoderVP9::ensureContext(const Vp9FrameHdr* hdr)
 {
+    bool resetContext = false;
+    Decode_Status status = DECODE_SUCCESS;
+    DEBUG("got frame size: %d x %d", hdr->width, hdr->height);
+
     // only reset va context when there is a larger frame
     if (m_configBuffer.width < hdr->width
         || m_configBuffer.height <  hdr->height) {
+        resetContext = true;
         INFO("frame size changed, reconfig codec. orig size %d x %d, new size: %d x %d",
                 m_configBuffer.width, m_configBuffer.height, hdr->width, hdr->height);
-        Decode_Status status = VaapiDecoderBase::terminateVA();
+        
+        if (m_hasContext){
+            status = VaapiDecoderBase::terminateVA();
+        }
+        m_hasContext = false;
         if (status != DECODE_SUCCESS)
             return status;
         m_configBuffer.width = hdr->width;
         m_configBuffer.height = hdr->height;
         m_configBuffer.surfaceWidth = hdr->width;
         m_configBuffer.surfaceHeight = hdr->height;
+        DEBUG("USE_NATIVE_GRAPHIC_BUFFER: %d",
```

Comment:
USE_NATIVE_GRAPHIC_BUFFER  is inherit from libmix, we will drop it latter. No need set this here. see #177 for details.